src/main/java/io/jcoder/tutorials/ch12/VendingMachine.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 - JCoder Ltd
+ */
+package io.jcoder.tutorials.ch12;
+
+/**
+ * Class used in the Vending Machine Project.
+ * 
+ * <p>
+ * https://jcoder.io/content/course/java/beginners/ch12/
+ * </p>
+ * 
+ * @author Andres Ricardo Perez
+ */
+public class VendingMachine {
+
+    private final ProductCatalog catalog;
+    private double maxAllowedBalance = 1000.00;
```

Comment:
We can change this to be a constant now. In chapter 4 I didn't make this a constant as we hadn't introduced static members, but now we can use them.

```suggestion
    private static final double MAX_ALLOWED_BALANCE = 1000.00;
```